x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccountsTests.java

```diff
@@ -46,6 +145,280 @@
 
 public class ElasticServiceAccountsTests extends ESTestCase {
 
+    public void testKibanaSystemPrivileges() {
```

Comment:
This works, but might not be necessary and also add to ongoing maintanence since the same is tested in `ReservedRoleStoreTests`. I'd prefer to just assert the service account role descriptor is equivalent to the one used for the reserved role. Something like:
```
var serviceRoleDescriptor = ElasticServiceAccounts.ACCOUNTS.get("elastic/kibana-system").roleDescriptor();
var reservedRoleDescriptor = ReservedRoleStore.roleDescriptor(KibanaSystemUser.ROLE_NAME);
assertThat(serviceRoleDescriptor.getClusterPrivileges(), equalTo(reservedRoleDescriptor.getClusterPrivileges()));
...
```
One caveat of the above is that the method `ReservedRoleStore.roleDescriptor(...)` is currently not `static`. But I think it should be OK to change it to `static` or alternatively we can use `new ReservedRolesStore().roleDescriptor(...)` similar to how it is done in `ReservedRoleStoreTests`.

Reply:
Agreed, I [was also not particularly fond](https://github.com/elastic/elasticsearch/pull/76370#discussion_r686983690) of this, your idea makes sense, thanks!